File: TemperatureApp/TemperatureApp/ViewController.m
//
//  ViewController.m
//  TemperatureApp
//
//  Created by pavan krishna on 22/02/17.
//  Copyright © 2017 KSU. All rights reserved.
//

#import "ViewController.h"
#import "AFNetworking.h"
#import <AFHTTPSessionManager.h>

static NSString * const kAPIUrl = @"http://api.openweathermap.org/data/2.5/weather?id=4276614&appid=0f7e07a9d99396dcf15d793331c720b8";

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    int result = 0;
    
    result = 2;
    
//    [self.temperatureLabel setText:@(result).stringValue];
//    [self.temperatureLabel setText:[NSString stringWithFormat:@"%d",result]];

    // Do any additional setup after loading the view, typically from a nib.
    [self doApiCall];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

-(void)doApiCall;
{

    // 2.0 code
//    NSURL *url = [NSURL URLWithString:kAPIUrl];
//    NSURLRequest *request = [NSURLRequest requestWithURL:url];
//    
//    AFHTTPRequestOperation *operation = [[AFHTTPRequestOperation alloc] initWithRequest:request];
//    operation.responseSerializer = [AFJSONResponseSerializer serializer];
//    
//    [operation setCompletionBlockWithSuccess:^(AFHTTPRequestOperation *operation, id responseObject) {
//        
//        NSLog(@"responseObject:%@",responseObject);
//        
//        NSDictionary *mainDictionary = [responseObject objectForKey:@"main"];
//        
//        NSString *temperatureinCelsius = [NSString stringWithFormat:@"%@ºc",[mainDictionary objectForKey:@"temp"]];
//        
//        [self.temperatureLabel setText:temperatureinCelsius];
//
//    } failure:^(AFHTTPRequestOperation *operation, NSError *error) {
//        
//        NSLog(@"error:%@",error);
//
//        UIAlertView *alertView = [[UIAlertView alloc] initWithTitle:@"Error Retrieving Data"
//                                                            message:[error localizedDescription]
//                                                           delegate:nil
//                                                  cancelButtonTitle:@"Ok"
//                                                  otherButtonTitles:nil];
//        [alertView show];
//    }];
//    
//    [operation start];
    
    
    // 3.0 code
    
    AFHTTPSessionManager *manager = [AFHTTPSessionManager manager];
    
    NSDictionary *params = @{@"email": @"dilip@kent.edu",
                             @"password": @"123456"};
    
    [manager POST:@"http://parkapps.kent.edu/ksu-cco/ios/login_check.php" parameters:params constructingBodyWithBlock:^(id<AFMultipartFormData>  _Nonnull formData) {
        
    } success:^(NSURLSessionDataTask * _Nonnull task, id  _Nonnull responseObject) {
         NSLog(@"JSON: %@", responseObject);
//        [[self navigationController] pushViewController:<#(nonnull UIViewController *)#> animated:<#(BOOL)#>]
    } failure:^(NSURLSessionDataTask * _Nullable task, NSError * _Nonnull error) {
        NSLog(@"Error: %@", error);
    }];
    
    
//    AFHTTPSessionManager *manager = [AFHTTPSessionManager manager];
//    
//    
//    [manager GET:kAPIUrl parameters:nil success:^(NSURLSessionDataTask * _Nonnull task, id  _Nonnull responseObject) {
//        
//                NSLog(@"responseObject:%@",responseObject);
//
//    } failure:^(NSURLSessionDataTask * _Nullable task, NSError * _Nonnull error) {
//        
//                NSLog(@"error:%@",error);
//        
//    }];
}

-(void)parseJSONSData;
{
    
}




@end

File: SortingAlgorithms/SortingAlgorithms/ViewController.swift
//
//  ViewController.swift
//  SortingAlgorithms
//
//  Created by pavan krishna on 26/08/17.
//  Copyright © 2017 pavan krishna. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
//        bubbleSort(numbersArray)
        selectionSort(numbersArray)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    var numbersArray = [14,33,27,10,35,19,42,44]
    
    func bubbleSort(_ numArray :[Int]) {
        
        // Worst complexity : O(n2)
        // Average complexity : O(n2)
        
        var tempArray = numArray
        
        for j in 0...tempArray.count-1 {
            for i in 0...tempArray.count-2 {
                if tempArray[i] > tempArray[i+1] {
                    swap(&tempArray[i], &tempArray[i+1])
                }
            }
        }
        
        for i in 0...tempArray.count-1 {
            print(tempArray[i])
        }
    }
    
    func insertionSort(_ numArray :[Int]) {
        
        // Worst complexity : O(n2)
        // Average complexity : O(n2)
        
        // first take the elements and add them to new array based on sorting
        var tempArray = numArray
        var newArray = [Int]()
        
        for i in 0...tempArray.count-1 {
            print(tempArray[i])
            newArray.append(tempArray[i])
        }
        
        for j in 0...newArray.count-1 {
            print(newArray[j])
        }
        
    }
    
    func selectionSort(_ numArray:[Int]) {
        // Worst complexity : O(n2)
        // Average complexity : O(n2)
        
        var tempArray = numArray
        var minIndex = 0

        // first take the first element and find the element which is less than first element and replace it the lowest element
        
        
        for i in 0...tempArray.count-1 {
            
            minIndex = i
            
            for j in i+1...tempArray.count-1 {
                
                if tempArray[j] < tempArray[minIndex] {
                    minIndex = j
                }
            }
            if minIndex != i {
                swap(&tempArray[minIndex], &tempArray[i])
            }
            
        }

        for i in 0...tempArray.count-1 {
            print(tempArray[i])
        }
    }
    
    func mergeSort() {
        
    }
    
    func shellSort() {
        
    }
    
    func quickSort() {
        
    }
}

